### PR TITLE
Add cypress tutorial

### PR DESCRIPTION
**Pull request by bidoubiwa**

### CODE REVIEW

File: guides/cypress.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+If you have the following error: 
+```
+error cypress@7.4.0: The engine "node" is incompatible with this module. Expected version ">=12.0.0". Got "10.22.1"
+```
+It means that you are probably using node version `10`. Please upgrade to at least `v12` as [node 10 not maintained anymore](https://endoflife.date/nodejs). 
+
+
+## Setup Cypress
+
+The tests are looked for by Cypress in the following sub-folder `cypress/fixtures`. So let us create the sub-folders and our first test file.
```
Comment (bidoubiwa) [inline guides/cypress.md:54]:
```suggestion
The tests are looked for by Cypress in the following sub-folder `cypress/integration`. So let us create the sub-folders and our first test file. 
```

File: guides/cypress.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+## Create Tests 
+
+Lets open `first_tests.specs.js` to create some tests. 
+
+```js
+const HOST = `http://localhost:8080` // please adjust to correct app port
+
+describe(`My first test`, () => {
+  before(() => {
+    cy.visit(host) // Visit our app
```
Comment (bidoubiwa) [inline guides/cypress.md:75]:
```suggestion
    cy.visit(HOST) // Visit our app
```

File: guides/cypress.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+## How Does Cypress Work
+
+Cypress runs your tests in all the specified browsers. Both in headless mode (without a browser opening) and an open mode so you can observe why tests are failing in the browser itself.
+
+This is the none-headless mode:
+
+![Cypress tests](../assets/gifs/cypress.gif)
+
+
+
```
Comment (bidoubiwa) [inline guides/cypress.md:26]:
```suggestion
```

**Review by curquiza — CHANGES_REQUESTED** [review]
Awesome!!

I did some really basic requests for change.

I had an issue when using cypress following the tuto, but @bidoubiwa and I checked it and fixed it together during a call. So I don't add it to this review 🙂

⚠️ I found out many whitespaces at the end of the lines, can you check they are removed?
I will add an editorconfig file anyway to avoid this problem again 🙂

File: guides/cypress.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      - uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: cypress-videos
+          path: cypress/videos
+```
+
+## Conclusion
+
+Testing in front-end became a more accessible task. We should not hesitate to add tests as they avoid losing **a lot of time** on the long run.
```
[the remark this answers is not in the document]
Comment (curquiza) [reply guides/cypress.md:200]:
```suggestion
Testing in front-end became a more accessible task. We should not hesitate to add tests as they avoid losing **a lot of time** in the long run 🏃‍♀️
```

**Review by curquiza — CHANGES_REQUESTED** [review]

File: guides/cypress.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+## How Does Cypress Work
+
+Cypress runs your tests in all the specified browsers. Both in headless mode (without a browser opening) and an open mode so you can observe why tests are failing in the browser itself.
+
+This is the none-headless mode:
+
+![Cypress tests](../assets/gifs/cypress.gif)
+
+## Requirements:
```
Comment (curquiza) [inline guides/cypress.md:25]:
```suggestion
## Requirements
```

File: guides/cypress.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+This will open the following dashboard.
+
+![Cypress dashboard](../assets/screenshots/dashboard.png)
+
+Click on the file name `first_tests.specs.js`. This should run the tests: 
+
+![Cypress simple test](../assets/gifs/simple_test.gif)
+
+The tests will run again on every change in your code (watch mode).
```
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply guides/cypress.md:103]: it wasn't haha

File: guides/cypress.md [hunk 7]
```diff
@@ -116,6 +116,26 @@ it('Contains title', () => {
 })
 ```
 
+## Running Server and Tests in One Command
+
+A limitation of the previous system is that we have to both start our app at one side and the tests at the other. 
+A solution is to launch both with the same command using [start-server-and-test](https://www.npmjs.com/package/start-server-and-test).
+This package will start your app, wait for it to be fully running, and then run the tests.
```
Comment (curquiza) [inline guides/cypress.md:123]:
```suggestion
A limitation of the previous system is that we have to both start our app on one side and the tests on the other.<br>
A solution is to launch both with the same command using [start-server-and-test](https://www.npmjs.com/package/start-server-and-test).<be>
This package will start your app, wait for it to be fully running, and then run the tests.
```

File: guides/cypress.md [hunk 8]
```diff
@@ -116,6 +116,26 @@ it('Contains title', () => {
 })
 ```
 
+## Running Server and Tests in One Command
+
+A limitation of the previous system is that we have to both start our app at one side and the tests at the other. 
+A solution is to launch both with the same command using [start-server-and-test](https://www.npmjs.com/package/start-server-and-test).
+This package will start your app, wait for it to be fully running, and then run the tests. 
+
+The following command will run the tests in headless mode once the server is running. When the tests are done, the process is killed.
```
Comment (curquiza) [inline guides/cypress.md:125]:
```suggestion
The following command will run the tests in headless mode once the server is running. When the tests are done, the process is killed.
```

**Review by curquiza — CHANGES_REQUESTED** [review]

File: guides/cypress.md [hunk 9]
```diff
@@ -118,11 +118,11 @@ it('Contains title', () => {
 
 ## Running Server and Tests in One Command
 
-A limitation of the previous system is that we have to both start our app at one side and the tests at the other. 
-A solution is to launch both with the same command using [start-server-and-test](https://www.npmjs.com/package/start-server-and-test).
-This package will start your app, wait for it to be fully running, and then run the tests. 
+A limitation of the previous system is that we have to both start our app on one side and the tests on the other.<br>
+A solution is to launch both with the same command using [start-server-and-test](https://www.npmjs.com/package/start-server-and-test).<be>
```
Comment (curquiza) [inline guides/cypress.md:122]:
```suggestion
A solution is to launch both with the same command using [start-server-and-test](https://www.npmjs.com/package/start-server-and-test).<br>
```

File: guides/cypress.md [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+```
+# Cypress
+cypress/screenshots
+cypress/videos
+cypress/support
+cypress/plugins
+cypress/fixtures
+```
+
+### CI Implementation with GH Actions
```
Comment (curquiza) [inline guides/cypress.md:181]:
```suggestion
## CI Implementation with GH Actions
```

**Review by curquiza — APPROVED** [review]
🚀

**Comment by curquiza** [comment]
you can merge when you want @bidoubiwa 
PS: bors is not present in this repo 😂

**Pull request merged**